### PR TITLE
Add devcontainer Docker image

### PR DESCRIPTION
**Pull request by maresb**
**What is this PR about?**

Make GitPod (and VS Code) more usable by providing a pre-compiled devcontainer.

Test this via https://gitpod.io/#https://github.com/maresb/pymc

**Checklist**
+ [ ] Explain important implementation details 👆
+ [ ] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [ ] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [ ] Are the changes covered by tests and docstrings?
+ [ ] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## New features
- Base image from [mamba-org/micromamba-devcontainer](https://github.com/mamba-org/micromamba-devcontainer)
- Performs an editable install of the current pymc and sets up pre-commit hooks
- Adds a GitHub workflow to generate a new devcontainer image

## Bugfixes
- ...

## Documentation
- ...

## Maintenance
- ...

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -1,20 +1,39 @@
-image: ghcr.io/conda/conda-ci:main-linux-python3.9
+image: ghcr.io/maresb/pymc-devcontainer:latest
```
Comment (twiecki) [inline .gitpod.yml:1]: can we use a pymc channel here?

Comment (maresb) [reply .gitpod.yml:1]: Yes, but it needs to exist first. To create it, it should be sufficient to merge this, then run the "create devcontainer" GitHub Action (via the workflow dispatch button). Then we can change the channel here.

But this is only needed when changing the dependencies, so for the moment there is no problem to keep my channel.

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
Is this done?

**Comment by maresb** [comment]
"Done" is perhaps too generous.

It shouldn't break anything, and it should hopefully move the GitPod situation substantially forward.

I think we can squash-merge, try it out, and fix any problems we experience.

**Pull request merged**